rts/idris_utf8.h

```diff
@@ -1,24 +1,38 @@
 #ifndef _IDRIS_UTF8
 #define _IDRIS_UTF8
 
+#include <stdint.h>
+#include <stdlib.h>
+
 /* Various functions for dealing with UTF8 encoding. These are probably
    not very efficient (and I'm (EB) making no guarantees about their
    correctness.) Nevertheless, they mean that we can treat Strings as
    UFT8. Patches welcome :). */
 
+typedef uint8_t utf8_byte;
+
```

Comment:
not a fan of this typedef. uint8_t is always the correct type for it, and nobody will be confused by utf8 being represented by an array of uint8_t.